post.md

```diff
@@ -0,0 +1,226 @@
+# Robot Framework - using Request library for control RPI GPIO's
+
+Requests library is one of the most popular libraries implemented for Robot
+Framework. It is very important for testing scattered applications, so this is
+the first library I got to know in the Robot Framework. The library requires 
+installing `robotframework-requests` and `requests` for proper working. 
+`Collections` library is a standard built-in library.
+
+For checking `Request` library we can install Pi-GPIO-Server on Raspberry Pi.
+Server is able to control state of input and output of GPIOS using Rest API.
+
+Project is availiable on link: [Project weekend - GPIO Server](https://github.com/projectweekend/Pi-GPIO-Server)
+
+In the Rest Api we use methods:
+- Delete
+- Get
+- Options
+- Post
+- Put
+- Patch
+
+To control GPIO's on the RPI we need only `Get` and `Patch`, other methods are
+used in the same way - `Options` are exceptions.
+
+I assume you have already installed Robot Framework. If response is "NO",
+I would like invite you to read the installation paragraph
+[Installation paragraph](http://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#installation-instructions)
+It is official documentation for Robot Framework. Next we can install libraries.
+
+Use below commands to install libs:
+
+```
+apt-get install python-pip
+pip install -U requests
+pip install -U robotframework-requests
+```
+
+Now we can prepare RPI image. Install `RASPBIAN STRETCH WITH DESKTOP`. It is
+very good solution, because we can controll also GPIO's using buttons.
+
+![](http://3mdeb.com/wp-content/uploads/2017/10/web_browser_control.png)
+
+I hope that instruction of installation process is exact, so I will not duplicate
+the description. Take a few minutes for preparation of configurations for pins
+and restart gpio server:
+
+```
+sudo service gpio-server restart
+```
+
+For running server go to Pi-GPIO-Server and type:
+
+```
+python pi_gpio_server.py
+```
+
+Now, we can start writting code for control GPIO using Rest API and Robot Framework.
+
+```
+*** Settings ***
+
+*** Variables ***
+
+*** Test Cases ***
+
+*** Keywords ***
+```
+
+Run RPi and check its IP address. We will use IP for connection with API server.
+My RPI has IP: 192.168.0.46. Documentation says that server runs on `5000` port. 
+So we can prepare URL variable. On `Variables` section type:
+
+```
+${URL} http://192.168.0.46:5000
+```
+
+Let's test out first test `Get signals`. Firstly  we should create session 
+for request control in Robot Framework. This convention uses all
+libraries eg. SSH.
+
+To create session we use keyword `Create session` with parameters `alias`, `URl`
```

Comment:
URL

Reply:
Some more explanation what means what or at least link to documentaiton would be great

Reply:
a session